listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -213,12 +214,32 @@ def user_feed(user_name: str):
     notification_events = get_notification_events(user, count)
 
     recording_pin_events = get_recording_pin_events(
+        user=user,
         users_for_events=users_for_feed_events,
         min_ts=min_ts or 0,
         max_ts=max_ts or int(time.time()),
         count=count,
     )
 
+    hidden_events = db_user_timeline_event.get_hidden_timeline_events(user['id'], count)
+    hidden_events_pin = {}
+    hidden_events_recommendation = {}
+
+    for hidden_event in hidden_events:
+        if hidden_event.event_type.value == UserTimelineEventType.RECORDING_RECOMMENDATION.value:
+            # hidden_events_recommendation.append(hidden_event)
+            hidden_events_recommendation[hidden_event.event_id] = hidden_event
+        else:
+            hidden_events_pin[hidden_event.event_id] = hidden_event
+
+    for i in range(len(recording_recommendation_events)):
+        if recording_recommendation_events[i].id in hidden_events_recommendation:
+            recording_recommendation_events[i].hidden = True
+
+    for i in range(len(recording_pin_events)):
```

Comment:
probably rewrite it as `for event in recording_pin_events:` and then `event.hidden = True` inside.

Reply:
Same for above loop.

Reply:
hi @amCap1712, if that is done, the changes wouldn't reflect in `recording_pin_events` list. Which is later being served in response. It would only change the event object in the current loop.

Reply:
If you use `for event in recording_pin_events`, the variable event in the current loop will refer to the original object in the list so it should work fine.

For example, try running this snippet of code and see how the value changes even if we do it this way.

```python
class Container:
    
    def __init__(self, value):
        self.value = value
        
    def __repr__(self):
        return f"Container[{self.value}]"
    
    def __str__(self):
        return f"Container[{self.value}]"
        
items = [Container(i) for i in range(1, 4)]
print(items)

for item in items:
    item.value = item.value * 5
    
print(items)
```

Reply:
Oops, I see, I didn't know objects behaved like that. I was merely thinking of them as regular datatypes in a list. Thank you so much @amCap1712 for taking your time to explain this. I have committed according to your review.

Once again thanks a lot!